app/components/UI/Bridge/hooks/useSwapBridgeNavigation/index.ts

```diff
@@ -0,0 +1,137 @@
+import { useCallback } from 'react';
+import { useNavigation } from '@react-navigation/native';
+import useGoToPortfolioBridge from '../useGoToPortfolioBridge';
+import { isBridgeUiEnabled } from '../../utils';
+import Routes from '../../../../../constants/navigation/Routes';
+import { Hex } from '@metamask/utils';
+import Engine from '../../../../../core/Engine';
+import { useSelector } from 'react-redux';
+import { selectChainId } from '../../../../../selectors/networkController';
+import { BridgeToken, BridgeViewMode } from '../../types';
+import { getNativeAssetForChainId } from '@metamask/bridge-controller';
+import { BridgeRouteParams } from '../useInitialSourceToken';
+///: BEGIN:ONLY_INCLUDE_IF(keyring-snaps)
+import { SolScope } from '@metamask/keyring-api';
+///: END:ONLY_INCLUDE_IF
+
+/**
+ * Returns functions that are used to navigate to the MetaMask Bridge and MetaMask Swaps routes.
+ * @param location location of navigation call – used for analytics.
+ * @param token token object containing address and chainId.
+ * @returns An object containing functions that can be used to navigate to the existing Bridges page in the browser and the MetaMask Swaps page. If there isn't an existing bridge page, one is created based on the current chain ID and passed token address (if provided).
+ */
+export const useSwapBridgeNavigation = ({
+  location,
+  sourcePage,
+  token: tokenBase,
+}: {
+  location: string;
```

Comment:
nit: If location is going to be used for analytics I'd recommend using a defined set of values possible, for example 
```suggestion
  location: 'TabBar' | 'TokenDetails';
```
or an enum if possible, in case there's a new location this should be added to the type and of course let the relevant teams know about this to be added in the analytics dashboard.

Reply:
This was just taken from the original code, so I didn't change anything and just passed in raw strings as it was before.